pkg/kubelet/volumemanager/reconciler/reconstruct_new.go

```diff
@@ -180,6 +180,14 @@ func (rc *reconciler) cleanOrphanVolumes() {
 func (rc *reconciler) updateReconstructedDevicePaths() {
 	klog.V(4).InfoS("Updating reconstructed devicePaths")
 
+	if rc.kubeClient == nil {
+		// Skip reconstructing devicePath from node objects if kubelet is in standalone mode.
+		// Such kubelet is not expected to mount any attachable volume or Secrets / ConfigMap.
+		klog.V(2).InfoS("Skipped reconstruction of DevicePaths from node.status in standalone mode")
+		rc.volumesNeedDevicePath = nil
+		return
+	}
+
 	node, fetchErr := rc.kubeClient.CoreV1().Nodes().Get(context.TODO(), string(rc.nodeName), metav1.GetOptions{})
```

Comment:
There are three modes the kubelet can be in:
1. never talk to an API server (kubeClient is nil)
2. talk to an API server it is responsible for starting (kubeClient is non-nil, but API calls will fail until the kubelet starts its static pods)
3. talk to an API server someone else is responsible for starting (kubeClient is non-nil, API calls will ~eventually work if the kubelet just keeps on trying)

This PR is working to fix scenario 1.

In scenario 2, does `updateReconstructedDevicePaths()` block the kubelet from running the kube-apiserver container?

Reply:
2. should be tested in CI, because all `-kind-` jobs use static pods

In pull-kubernetes-e2e-kind I can see in [one of the kubelets](https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/116577/pull-kubernetes-e2e-kind/1635605473094799360/artifacts/kind-control-plane/kubelet.log) there:

`
Mar 14 11:48:18 kind-control-plane kubelet[288]: I0314 11:48:18.354841     288 kubelet.go:1821] "Creating a mirror pod for static pod" pod="kube-system/kube-apiserver-kind-control-plane"
Mar 14 11:48:18 kind-control-plane kubelet[288]: I0314 11:48:18.354976     288 kubelet.go:1821] "Creating a mirror pod for static pod" pod="kube-system/kube-controller-manager-kind-control-plane"
...
`

Reply:
I didn't know if volume reconstruction took a different code path on restart of a kubelet with existing volume state on disk vs first-start

Reply:
If the call fails here, it will still proceed with the rest of the volume manager reconciler: https://github.com/kubernetes/kubernetes/blob/a9008b502d32a5dbacf835adb040419bf31f5c04/pkg/kubelet/volumemanager/reconciler/reconciler_new.go#L60

The device path information is only needed for attachable volume types, which is not supported in standalone mode. It would still be good to test the reconstruction scenarios for the different scenarios.